AutomatedTesting/Gem/PythonTests/Atom/atom_utils/material_canvas_utils.py

```diff
@@ -10,101 +10,101 @@
-        source_node, source_slot, target_node, target_slot)
+class Graph(object):
+    """
+    Opens the specified materialgraph file when inside MaterialCanvas which appears as a Graph.
+    Opening the Graph is required since it also sets the document_id for use with other functions.
+    """
+
+    def __init__(self, material_graph_file_path: str):
+        self.material_graph_file_path = material_graph_file_path
+        self.document_id = atom_tools_utils.open_document(self.material_graph_file_path)
+        self.graph_id = atomtools.GraphDocumentRequestBus(
+            bus.Event, GraphDocumentRequestBusEvents.GET_GRAPH_ID, self.document_id)
+        self.name = atomtools.GraphDocumentRequestBus(
+            bus.Event, GraphDocumentRequestBusEvents.GET_GRAPH_NAME, self.document_id)
+        self.graph_object = atomtools.GraphDocumentRequestBus(
+            bus.Event, GraphDocumentRequestBusEvents.GET_GRAPH, self.document_id)
+        self.nodes = []
+
+    def _create_node_by_name(self, node_name: str) -> object:
+        """
+        Creates a new node in memory matching the node_name string on the specified graph object.
+        i.e. "World Position" for node_name would create a World Position node.
+        :param node_name: String representing the type of node to add to the graph.
+        :return: azlmbr.object.PythonProxyObject representing a C++ AZStd::shared_ptr<Node> object.
+        """
+        return atomtools.DynamicNodeManagerRequestBus(
+            bus.Broadcast, DynamicNodeManagerRequestBusEvents.CREATE_NODE_BY_NAME, self.graph_object, node_name)
+
+    def add_node(self, node_name: str, position: math.Vector2) -> None:
+        """
+        Creates a node in memory using _create_node_by_name then adds it to this Graph object.
+        :param node_name: String representing the type of node to add to the graph.
+        :param position: math.Vector2(x,y) value that determines where to place the node on the Graph's grid.
+        :return: None, but appends the new node to self.nodes list.
+        """
+        node_object = self._create_node_by_name(node_name)
+        node_id = graph.GraphControllerRequestBus(
+            bus.Event, GraphControllerRequestBusEvents.ADD_NODE, self.graph_id, node_object, position)
+        self.nodes.append(Node(node_object, node_id, node_name, self.graph_id, position))
+
+
+class Node(object):
+    """
+    Represents a node inside of a Graph (required).
+    """
+
+    def __init__(self, node_object: object, node_id: int, node_name: str, graph_id: math.Uuid, position: math.Vector2):
+        self.node_object = node_object
+        self.node_id = node_id
+        self.node_name = node_name
+        self.graph_id = graph_id
+        self.position = position
+        self.outbound_slot = graph.GraphModelSlotId('outPosition')
+        self.inbound_slot = graph.GraphModelSlotId('inPositionOffset')
+
+    def connect_to_node_outbound_slot(self, target_node: Node) -> object:
```

Comment:
could be slightly clearer function name
def connect_inbound_to_target_outbound_slot(self, target_node: Node, target_slot: SlotId
or
def connect_to_target_node(self, self.Slot, target_node: Node, target_slot: SlotId

this might do well with a helper for GraphModelSlotId common name. Since each node has one or more inbound and outbound slots it may be unwieldy to just simply connect two nodes.